practice-app/requirements.txt

```diff
@@ -0,0 +1,18 @@
+altgraph @ file:///System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot7/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/python3/python3-133.100.1.1/altgraph-0.17.2-py2.py3-none-any.whl
```

Comment:
This dependency logic does not work as it searches the local drive, rather then the PyPi. Thus, project setup will not work in others.

It should be in the form of
altgraph==x.y.z (0.17.2, for this case)

Reply:
To elaborate a bit more:

When we use `some_package==0.3.5`, pip automatically searches for the package in the pypi.org (if you want to look, pypi.org/simple), then finds the package, then downloads the package.

However, it's not necessary that we're going to use a package from PyPi. In fact, we may have our own package somewhere in our computer (or a modification of a package in somewhere), and we may want to use it. 

To do so, one can simple add `@` and the insert the path that pip installer should search for that package.

In our example, pip looks for the `altgraph` package in the filesystem of the computer, specifically looks at the `system/volumes/.... path`. If it cannot find it, it throws an error. For this case, it's probably in someone's computer, but not mine (and probably most of us do not have that package in that location).